server/channels/app/plugin_api.go

```diff
@@ -1262,3 +1263,34 @@ func (api *PluginAPI) GetUploadSession(uploadID string) (*model.UploadSession, e
 	}
 	return fi, nil
 }
+
+// SendPushNotification will attempt to send a push notification to `notification.User`, using
+// `notification.Post` as the source of the notification. Note: the NotificationWillBePushed hook will
+// be run after SendPushNotification is called.
+func (api *PluginAPI) SendPushNotification(notification *model.PluginPushNotification) error {
```

Comment:
This adds it to the `app` side of the interface, but I don't see the corresponding changes to the RPC layer and interface that make it actually work with a real plugin. Is that somehow separate from this PR?